br/pkg/task/encryption.go

```diff
@@ -0,0 +1,169 @@
+	case SchemeGCP:
+		return parseGcpKmsConfig(u)
+	default:
+		return encryptionpb.MasterKey{}, errors.Errorf("unsupported master key type: %s", u.Scheme)
+	}
+}
+
+func parseLocalDiskConfig(u *url.URL) (encryptionpb.MasterKey, error) {
+	if !path.IsAbs(u.Path) {
+		return encryptionpb.MasterKey{}, errors.New("local master key path must be absolute")
+	}
+	return encryptionpb.MasterKey{
+		Backend: &encryptionpb.MasterKey_File{
+			File: &encryptionpb.MasterKeyFile{
+				Path: u.Path,
+			},
+		},
+	}, nil
+}
+
+func parseAwsKmsConfig(u *url.URL) (encryptionpb.MasterKey, error) {
+	matches := awsRegex.FindStringSubmatch(u.Path)
+	if matches == nil {
+		return encryptionpb.MasterKey{}, errors.New("invalid AWS KMS key ID format")
+	}
+	keyID := matches[1]
+
+	q := u.Query()
+	region := q.Get(AWSRegion)
+	accessKey := q.Get(AWSAccessKeyId)
+	secretAccessKey := q.Get(AWSSecretKey)
+
+	if region == "" {
+		return encryptionpb.MasterKey{}, errors.New("missing AWS KMS region info")
+	}
+
+	var awsKms *encryptionpb.AwsKms
+	if accessKey != "" && secretAccessKey != "" {
+		awsKms = &encryptionpb.AwsKms{
+			AccessKey:       accessKey,
+			SecretAccessKey: secretAccessKey,
+		}
+	}
+
+	return encryptionpb.MasterKey{
+		Backend: &encryptionpb.MasterKey_Kms{
+			Kms: &encryptionpb.MasterKeyKms{
+				Vendor:   AWSVendor,
+				KeyId:    keyID,
+				Region:   region,
+				Endpoint: q.Get(AWSEndpoint), // Optional
+				AwsKms:   awsKms,             // Optional, can read from env
+			},
+		},
+	}, nil
+}
+
+func parseAzureKmsConfig(u *url.URL) (encryptionpb.MasterKey, error) {
+	matches := azureRegex.FindStringSubmatch(u.Path)
+	if matches == nil {
+		return encryptionpb.MasterKey{}, errors.New("invalid Azure KMS path format")
+	}
+
+	keyID := matches[1] // This now captures the entire path as the key ID
+	q := u.Query()
+
+	azureKms := &encryptionpb.AzureKms{
+		TenantId:     q.Get(AzureTenantID),
+		ClientId:     q.Get(AzureClientID),
+		ClientSecret: q.Get(AzureClientSecret),
+		KeyVaultUrl:  q.Get(AzureVaultName),
+	}
+
+	if azureKms.TenantId == "" || azureKms.ClientId == "" || azureKms.ClientSecret == "" || azureKms.KeyVaultUrl == "" {
+		return encryptionpb.MasterKey{}, errors.New("missing required Azure KMS parameters")
+	}
+
+	return encryptionpb.MasterKey{
+		Backend: &encryptionpb.MasterKey_Kms{
+			Kms: &encryptionpb.MasterKeyKms{
+				Vendor:   AzureVendor,
+				KeyId:    keyID,
+				AzureKms: azureKms,
+			},
+		},
+	}, nil
+}
+
+func parseGcpKmsConfig(u *url.URL) (encryptionpb.MasterKey, error) {
+	matches := gcpRegex.FindStringSubmatch(u.Path)
+	if matches == nil {
+		return encryptionpb.MasterKey{}, errors.New("invalid GCP KMS path format")
+	}
+
+	projectID, location, keyRing, keyName := matches[1], matches[2], matches[3], matches[4]
+	q := u.Query()
+	credentials := q.Get(GCPCredentials)
+
+	if credentials == "" {
+		return encryptionpb.MasterKey{}, errors.Errorf("missing required GCP KMS parameter: %s", GCPCredentials)
```

Comment:
I guess GCP SDK can also [read credentials from env](https://cloud.google.com/docs/authentication/application-default-credentials?hl=zh-cn)?

Reply:
ah you are right, good call!